core/src/main/java/org/owasp/dependencycheck/analyzer/PinnedMavenInstallAnalyzer.java

```diff
@@ -118,31 +121,57 @@ protected void analyzeDependency(Dependency dependency, Engine engine) throws An
         }
 
         final DependencyTree tree;
+        List<? extends HasCoord> deps = null;
         try {
-            final InstallFile installFile = INSTALL_FILE_READER.readValue(dependencyFile);
-            tree = installFile.getDependencyTree();
+            JsonNode jsonNode = MAPPER.readTree(dependencyFile);
+            JsonNode v2Version = jsonNode.path("version");
+            JsonNode v010Version = jsonNode.path("dependency_tree").path("version");
+
+            if (v2Version.isMissingNode() && v010Version.isMissingNode()) {
+                LOGGER.warn("No pinned maven_install.json version found. Cannot Parse");
+                return;
+            } else if (!v2Version.isMissingNode()) {
```

Comment:
🧠 ⚡ The double negative of `!v.isMissingNode()` is slightly tricky to read. What about `isTextual()`?

Or you could do:

```
if (!jsonNode.has("version") && !jsonNode.path("dependency_tree").has("version")) {
  // warn & return
} else if (jsonNode.has("version") {
  // etc
```

Not sure it's better -- but maybe.